smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/form/SoftwareInfoForm.java

```diff
@@ -0,0 +1,275 @@
+            for (MediaElement.Uri uri : uriList) {
+                mediaBuilder.addUri(uri);
+            }
+            MediaElement mediaElement = mediaBuilder.setHeightAndWidth(height, width).build();
+            return setIcon(mediaElement);
+        }
+
+        /**
+         * This will allow to include {@link MediaElement} directly as a
+         * {@link FormField}.
+         * <br>
+         * @param mediaElement MediaElement to be included
+         * @return Builder
+         */
+        public Builder setIcon(MediaElement mediaElement) {
+            FormField.Builder<?, ?> builder = FormField.builder("icon");
+            builder.addFormFieldChildElement(mediaElement);
+            dataFormBuilder.addField(builder.build());
+            return this;
+        }
+
+        /**
+         * Include Operating System's name as a {@link FormField}.
+         * <br>
+         * @param os Name of the OS
+         * @return Builder
+         */
+        public Builder setOS(String os) {
+            TextSingleFormField.Builder builder = FormField.builder("os");
+            builder.setValue(os);
+            dataFormBuilder.addField(builder.build());
+            return this;
+        }
+
+        /**
+         * Include Operating System's version as a {@link FormField}.
+         * <br>
+         * @param os_version Version of OS
+         * @return Builder
+         */
+        public Builder setOSVersion(String os_version) {
+            TextSingleFormField.Builder builder = FormField.builder("os_version");
+            builder.setValue(os_version);
+            dataFormBuilder.addField(builder.build());
+            return this;
+        }
+
+        /**
+         * Include Software name as a {@link FormField}.
+         * <br>
+         * @param software Name of the software
+         * @return Builder
+         */
+        public Builder setSoftware(String software) {
+            TextSingleFormField.Builder builder = FormField.builder("software");
+            builder.setValue(software);
+            dataFormBuilder.addField(builder.build());
+            return this;
+        }
+
+        /**
+         * Include Software Version as a {@link FormField}.
+         * <br>
+         * @param softwareVersion Version of the Software in use
+         * @return Builder
+         */
+        public Builder setSoftwareVersion(String softwareVersion) {
+            TextSingleFormField.Builder builder = FormField.builder("software_version");
+            builder.setValue(softwareVersion);
+            dataFormBuilder.addField(builder.build());
+            return this;
+        }
+
+        /**
+         * Include {@link DataForm} to be encapsulated under SoftwareInfoForm.
+         * <br>
+         * @param dataForm The dataform containing Software Information
+         * @return Builder
+         */
+        public Builder setDataForm(DataForm dataForm) {
+            if (dataForm.getTitle() != null || dataForm.getItems().size() != 0
```

Comment:
You have the habit of using `size() == 0` (or the inverse), when often `isEmpty()` would be more idiomatic (if availble).